### ISSUE TITLE
New API call for exposing YR_MATCHES

### ISSUE THREAD

**Issue by willgamz** [VirusTotal/yara#1586]
Hi

When implementing wrappers around libyara YR_MATCHES is only accessible using a macro: yr_string_matches_foreach
The YR_MATCHES values are under CONTEXT, and CONTEXT depends on the platform, should be accessed directly. 

This macro wont be available for example to a pure .Net wrapper,  forcing to have another "C/C++, etc" layer on top of libyara.
A simple API call make the wrappers dev much easier. For example:

YR_API YR_MATCHES* yr_scanner_get_matches(YR_SCANNER* scanner)
{
  return scanner->matches;
}

Thanks.

**Comment by plusvic** [comment]
But `yr_string_matches_foreach` allows iterating the matches for a specific string, while `yr_scanner_get_matches` would return all the matches for all the strings right? As the `YR_MATCH` struct doesn't have a pointer to the corresponding `YR_STRING` the user of the API doesn't have a way for knowing the which is the string for that match, unless they know some implementation details.

I guess the solution would be some kind of API that does the same thing than `yr_string_matches_foreach` but without using a macro.

**Comment by willgamz** [comment]
Yes, correct. For example, we can get the matches during callback, processing CALLBACK_MSG_RULE_MATCHING, we know the pointer to YR_STRING and we can get the all MATCHES for that YR_STRING. But you have to use the macros.

However, the idea is to implement the API for each of the MACROS, given those MACROS are not accessible outside of c,c++.
